device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -161,10 +156,6 @@ void open() throws IOException
         {
             throw new IOException("Could not open the connection", e);
         }
-
-        /* Codes_SRS_DEVICE_IO_21_014: [The open shall schedule receive tasks to run every receivePeriodInMilliseconds milliseconds.] */
-        /* Codes_SRS_DEVICE_IO_21_016: [The open shall set the `state` as `CONNECTED`.] */
-        commonOpenSetup();
```

Comment:
This logic was moved to the connection status callback. That way, this method (which starts up the send/receive thread pool) gets called every time the client goes from DISCONNECTED or DISCONNECTED_RETRYING to CONNECTED. This allows the thread pool to not be active during DISCONNECTED_RETRYING